src/puffin/src/puffin_manager/tests.rs

```diff
@@ -0,0 +1,361 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use common_test_util::temp_dir::{create_temp_dir, TempDir};
+use futures::AsyncReadExt as _;
+use tokio::fs::File;
+use tokio::io::AsyncReadExt as _;
+use tokio_util::compat::{Compat, TokioAsyncReadCompatExt};
+
+use crate::blob_metadata::CompressionCodec;
+use crate::error::Result;
+use crate::puffin_manager::cache_manager::MokaCacheManager;
+use crate::puffin_manager::cached_puffin_manager::CachedPuffinManager;
+use crate::puffin_manager::file_accessor::PuffinFileAccessor;
+use crate::puffin_manager::{
+    BlobGuard, DirGuard, PuffinManager, PuffinReader, PuffinWriter, PutOptions,
+};
+
+async fn new_moka_cache_manager(prefix: &str) -> (TempDir, Arc<MokaCacheManager>) {
+    let cache_dir = create_temp_dir(prefix);
+    let path = cache_dir.path().to_path_buf();
+    (
+        cache_dir,
+        Arc::new(MokaCacheManager::new(path, rand::random()).await.unwrap()),
+    )
+}
+
+#[tokio::test]
+async fn test_put_get_file() {
+    let compression_codecs = [None, Some(CompressionCodec::Zstd)];
+
+    for compression_codec in compression_codecs {
```

Comment:
non-blocking: Maybe we can predefine a list of capacities instead of a random capacity.

```rust
for capacity in capacities {
    for compression_codec in compression_codecs {
    }
}
```